pkgs/build-support/fetchurl/default.nix

```diff
@@ -160,5 +160,5 @@ stdenvNoCC.mkDerivation {
   '';
 
   inherit meta;
-  inherit passthru;
+  passthru = { inherit url; } // passthru;
```

Comment:
This seems like something to be discussed and not nonchalantly introduced updating a package.
I don't have a problem with it, but haven't really thought about it a lot.  Fetching `url` back does seem useful, and I see it helps a bit for how the devilutionx expression is structured.

Alternatively, it'd be easy enough to just introduce a helper function in devilutionx that either does effectively this (put `url` in `passthru`), or returns both the resulting store path and the fetched url together: `fetch = url: hash: { src = fetchurl { inherit url hash; }; inherit url; }; asio = fetch "https://..." "sha256-..."; ...` or maybe something little tidier :).  And in the substitutions each accesses `asio.url` and `asio.src`.

Anyway if fetchurl has a `.url` what you have sounds great, but `fetchurl` is not usually casually modified like this.

Reply:
Most other fetchers like fetchFromGitHub do similar things and the performance difference is negligible. You can already pass it through like ``fetchurl rec { passthru.url = url; }`` which is IMO a bit useless without a good default. Definitely changes nothing about how fetchurl works. Extracting it into another PR just creates overhead.